### PR TITLE
Batch write

### PR DESCRIPTION
**Pull request by zhexuany**
merge batch_write to master.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala [hunk 1]
```diff
@@ -261,15 +300,14 @@ object SharedSQLContext extends Logging {
...
-      sparkConf.set(ALLOW_INDEX_READ, getFlagOrTrue(prop, ALLOW_INDEX_READ).toString)
-      sparkConf.set(ENABLE_AUTO_LOAD_STATISTICS, "true")
-      sparkConf.set("spark.sql.decimalOperations.allowPrecisionLoss", "false")
-      sparkConf.set(REQUEST_ISOLATION_LEVEL, SNAPSHOT_ISOLATION_LEVEL)
-      sparkConf.set("spark.sql.extensions", "org.apache.spark.sql.TiExtensions")
+//      sparkConf.set(PD_ADDRESSES, getOrElse(prop, PD_ADDRESSES, "127.0.0.1:2379"))
+//      sparkConf.set(ENABLE_AUTO_LOAD_STATISTICS, "true")
+//      sparkConf.set("spark.sql.decimalOperations.allowPrecisionLoss", "false")
+//      sparkConf.set(REQUEST_ISOLATION_LEVEL, SNAPSHOT_ISOLATION_LEVEL)
+//      sparkConf.set("spark.sql.extensions", "org.apache.spark.sql.TiExtensions")
```
Comment (marsishandsome) [inline core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala]: remove

File: core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala [hunk 2]
```diff
@@ -280,6 +318,19 @@ object SharedSQLContext extends Logging {
 
       _tidbConf = prop
 
+      sparkConf.set(PD_ADDRESSES, pdAddresses)
```
Comment (marsishandsome) [inline core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala]: pls add a TODO here and create a issue

**Comment by zhexuany** [comment]
https://github.com/pingcap/tispark/issues/703 will be fixed in this PR.

**Pull request merged**